src/lib/executors/Node.ts

```diff
@@ -875,13 +876,34 @@ export default class Node extends Executor<NodeEvents, Config, NodePlugins> {
         }
 
         testTask
-          .then(() => {
+          .then(async () => {
             if (!this._sessionSuites) {
               return;
             }
-            return this._loadFunctionalSuites().then(
-              () => (testTask = this._runRemoteTests())
-            );
+
+            let remainingAttempts = 1 + (this.config.functionalRetries || 0);
+            let suites = this._sessionSuites;
+
+            await this._loadFunctionalSuites();
+            while (remainingAttempts && suites.length) {
+              remainingAttempts--;
+              if (suites.length !== this._sessionSuites!.length) {
+                this.log(
+                  'reattempting',
+                  suites.length,
+                  'of',
+                  this._sessionSuites!.length,
+                  'environments'
+                );
+              }
+              testTask = this._runRemoteTests(suites);
+              await testTask.catch(remainingAttempts ? () => {} : undefined);
```

Comment:
Why not a `try/catch`?

Reply:
This seemed less verbose than

Reply:
I think we should be more explicit in general; `catch(remainingAttempts ? () => {} : undefined)` obscures the fact that the goal is to consume or throw the error. If you expand the catch, it's not much different than the `try / catch`, so we should probably use that for consistency.